RenderSystems/GLSupport/CMakeLists.txt

```diff
@@ -116,6 +134,16 @@ target_include_directories(OgreGLSupport PUBLIC
     "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include/GLSL>"
     PRIVATE "$<BUILD_INTERFACE:${NATIVE_INCLUDES}>")
 
+  if(OGRE_GLSUPPORT_USE_WAYLAND)
+    target_compile_definitions(OgreGLSupport PUBLIC OGRE_WAYLAND)
+    # TODO should be part of include file?
+
+    add_dependencies(OgreGLSupport xdg_shell)
```

Comment:
why is xdg_shell needed?

Reply:
It was needed for fullscreen stuff and xdg_ data structures. Removed in new commit.